packages/datadog-plugin-cucumber/test/index.spec.js

```diff
@@ -0,0 +1,160 @@
+'use strict'
+
+const agent = require('../../dd-trace/test/plugins/agent')
+const plugin = require('../src')
+
+const {
+  TEST_FRAMEWORK,
+  TEST_TYPE,
+  TEST_NAME,
+  TEST_SUITE,
+  TEST_STATUS
+} = require('../../dd-trace/src/plugins/util/test')
+const { expect } = require('chai')
+const path = require('path')
+const { PassThrough } = require('stream')
+
+const TESTS = [
+  {
+    featureName: 'simple.feature',
+    featureSuffix: ':3',
+    requireName: 'simple.js',
+    testName: 'pass scenario',
+    testStatus: 'pass',
+    steps: [
+      { name: 'datadog', stepStatus: 'pass' },
+      { name: 'run', stepStatus: 'pass' },
+      { name: 'pass', stepStatus: 'pass' }
+    ],
+    success: true
+  },
+  {
+    featureName: 'simple.feature',
+    featureSuffix: ':8',
+    requireName: 'simple.js',
+    testName: 'fail scenario',
+    testStatus: 'fail',
+    steps: [
+      { name: 'datadog', stepStatus: 'pass' },
+      { name: 'run', stepStatus: 'pass' },
+      { name: 'fail', stepStatus: 'fail' }
+    ],
+    success: false,
+    errors: [undefined, undefined, 'AssertionError', 'AssertionError']
+  },
+  {
+    featureName: 'simple.feature',
+    featureSuffix: ':13',
+    requireName: 'simple.js',
+    testName: 'skip scenario',
+    testStatus: 'skip',
+    steps: [
+      { name: 'datadog', stepStatus: 'pass' },
+      { name: 'run', stepStatus: 'pass' },
+      { name: 'skip', stepStatus: 'skip' }
+    ],
+    success: true,
+    errors: [undefined, undefined, 'skipped', 'skipped']
+  },
+  {
+    featureName: 'simple.feature',
+    featureSuffix: ':19',
+    requireName: 'simple.js',
+    testName: 'skip scenario based on tag',
+    testStatus: 'skip',
+    steps: [{ name: 'datadog', stepStatus: 'skip' }],
+    success: true,
+    errors: ['skipped', 'skipped']
+  }
+]
+
+wrapIt()
+
+describe('Plugin', () => {
+  let Cucumber
+
+  withVersions(plugin, '@cucumber/cucumber', version => {
+    afterEach(() => {
+      // > If you want to run tests multiple times, you may need to clear Node's require cache
+      // before subsequent calls in whichever manner best suits your needs.
+      TESTS.forEach((test) => {
+        delete require.cache[require.resolve(path.join(__dirname, 'features', test.requireName))]
+      })
+      return agent.close()
+    })
+    beforeEach(() => {
+      return agent.load('cucumber').then(() => {
+        Cucumber = require(`../../../versions/@cucumber/cucumber@${version}`).get()
+      })
+    })
+
+    describe('cucumber', () => {
+      TESTS.forEach(test => {
+        it(`should create a test span for ${test.featureName}${test.featureSuffix || ''}`, async function () {
```

Comment:
Small nit, but that test is pretty long and tests many different things. It would benefit from being split up in multiple test cases.

Reply:
done in https://github.com/DataDog/dd-trace-js/pull/1251/commits/e549e1619b61d69c2ec50ced3b94c430a80cadc6